src/cpp/TextNoteCollection.h

```diff
@@ -16,6 +16,7 @@ class TextNoteCollection {
 	virtual ~TextNoteCollection();
 
 	TextNoteCollection* add(TextNote note);
+    TextNoteCollection* addFromCLI(const std::string& title);
```

Comment:
Maybe don't add this as a method of TextNoteCollection? This class could be used in context where no CLI is present, so it should not care about CLI?